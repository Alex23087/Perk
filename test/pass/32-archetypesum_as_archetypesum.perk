import "stdio.h"

archetype XDim {
  x: int
}

archetype YDim {
  y: int
}

model Point: XDim, YDim {
  let x := 0,
  let y := 0
}

fun main (): void {
  let p := summon Point();
  let xy := p as XDim+YDim;
  let x := xy ~> XDim;
  printf("X: %d\n", x.x);
  let y := xy ~> YDim;
  printf("Y: %d\n", y.y);

  y.y = 10;
  printf("Updated Y: %d\n", y.y);
  printf("Point: (%d, %d)\n", p.x, p.y);
  printf("Point as XYDim: (%d, %d)\n", xy.x, xy.y);
}