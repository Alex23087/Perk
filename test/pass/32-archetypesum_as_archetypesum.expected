X: 0
Y: 0
Updated Y: 10
Point: (0, 10)
Point as XYDim: (0, 10)
